crates/crates_io_tarball/src/builder.rs

```diff
@@ -47,3 +50,53 @@ impl AsMut<tar::Builder<Vec<u8>>> for TarballBuilder {
         &mut self.inner
     }
 }
+
+pub struct AsyncTarballBuilder {
```

Comment:
do we really need the async builder? is the sync one not enough for our test suite purposes?

Reply:
Indeed and fair enough! I implemented it without realizing it was primarily for testing 😅 
And since it's only for test purposes, I agree that we don't really need an async builder.

Reply:
7 changed files certainly looks easier to review 😂 

I'll take a look tomorrow :)